integr8ly/integreatly-operator
# INTLY-4237 - use Roles for backups

JIRA: https://issues.redhat.com/browse/INTLY-4237

Changes made in this PR:
- switching to Roles instead of ClusterRoles for backup ServiceAccount.
- syncing S3 backups secret from integreatly ns -> product namespaces that have backup jobs
- fixing postgres credentials secret for the backups
- addintg template and steps for providing S3 credentials into clusters that can't create S3 bucket automatically
- fixing a small issue in monitoring finalizer (a bit of offtopic, but it might give false negative result on e2e tests, so better fix it sooner rather than later). I've put it in a separate commit for easier review.

**Verification steps:**
Install Integreatly using this PR. Make sure to use ServiceAccount if running locally (`make code/run/service_account`).
Following newly added Readme steps, update the S3 credentials Secret with valid credentials.
Verify that existing backup jobs work correctly:
a) AMQ Online PV backups job:
- create a queue and some some data (you can follow WT1, ignoring integration part, just skip to Nodejs app and send some pineapples :) )
- trigger a Job from CronJob definition:
```
oc create job enmasse-pv-backup-$(date +"%Y-%m-%d-%H%M%S") --from=cronjob/enmasse-pv-backup -n rhmi-amq-online
```
- open `Workloads -> Jobs` in OCP console
- new Job will appear in the list, open it and navigatte to `Pods` tab
- open a Pod spawned by this job and watch the `Logs`
- there should be no errors in the logs, and the job should finish by uploading a backup artifact to S3 successfully
- Verify that a backup was successfully uploaded to your S3 bucket

b) CodeReady workspace PV backup job:
- login as test-user into CodeReady (if you see a Keycloak dialog asking you for email/name/etc., make sure integreatly operator is running, and go to CodeReady URL again)
- create a workspace and make sure workspace pod starts in codeready namespace (there is a bug with PVCs, you can try stopping the workspace in CRW UI and starting it again)
- Trigger a backup Job:
```
oc create job codeready-pv-backup-$(date +"%Y-%m-%d-%H%M%S") --from=cronjob/codeready-pv-backup -n rhmi-codeready-workspaces
```
- Verify the backup worked (see last 5 steps of the `a) AMQ Online PV backups job` steps)

c) CodeReady Postgres backup job:
- Trigger a backup Job:
```
oc create job codeready-postgres-backup-$(date +"%Y-%m-%d-%H%M%S") --from=cronjob/codeready-postgres-backup -n rhmi-codeready-workspaces
```
- Verify the backup worked (see last 5 steps of the `a) AMQ Online PV backups job` steps)
